app/assets/javascripts/Components/test_run_table.jsx

```diff
@@ -168,6 +168,10 @@ class TestGroupResultTable extends React.Component {
     };
   }
 
+  static getDerivedStateFromProps(props, _) {
```

Comment:
You've identified the correct problem (`filteredData` isn't reset), but it's not exactly correct to update it here. `filteredData` always needs to remain in sync with `filtered` so that the filters apply (and this can occur when the state changes). Instead, you can use `componentDidUpdate` and check whether `props.data` has changed, and if so, reset the filter entirely like what's done in the constructor.